src/scripts/DataLayerListenerManagerFactory.js

```diff
@@ -97,7 +98,13 @@ ListenerManagerFactory.create = function() {
         const listenerConfig = listener.config;
         const itemConfig = item.config;
         const itemConfigCopy = JSON.parse(JSON.stringify(itemConfig));
-        listenerConfig.handler(itemConfigCopy);
+        if (listener.config.selector) {
```

Comment:
can we just pass the state in this function? Would rather we keep the listener manager and data layer manager decoupled.

Reply:
we could. But if we want to move `_processListenerOn` and `_triggerListener` to `ListenerManagerFactory` (which I think we should), we would need to also pass `_dataLayer` to `ListenerManagerFactory.create()`.